lib/ecto/changeset.ex

```diff
@@ -1012,7 +1012,20 @@ defmodule Ecto.Changeset do
           "please use put_#{tag}/4 instead"
   end
 
-  defp put_change(data, changes, errors, valid?, key, value, _type) do
+  defp put_change(data, changes, errors, valid?, key, value, nil) do
```

Comment:
Oh yeah, good job me, keeping up with the issues 😆 

You're right - this is fine.